src/editors/EditorFooter.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useContext, useEffect } from 'react';
+import {
+  Spinner, ActionRow, Button, ModalDialog, Toast,
+} from '@edx/paragon';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import EditorPageContext from './EditorPageContext';
+import { ActionStates } from './data/constants';
+
+const navigateAway = (destination) => {
+  window.location.assign(destination); // do we want to do window replace?
+};
+
+export default function EditorFooter() {
+  const {
+    blockLoading,
+    setBlockContent,
+    unitUrlLoading,
+    unitUrl,
+    setSaveUnderway,
+    saveUnderway,
+    saveResponse,
+    studioEndpointUrl,
+    editorRef,
+  } = useContext(EditorPageContext);
+
+  const onSaveClicked = () => {
```

Comment:
These are easier to test if you define them outside the component in the same file :-)